lapce-data/src/hover.rs

```diff
@@ -64,14 +64,17 @@ impl HoverData {
             size: Size::new(600.0, 300.0),
             content_size: Rc::new(RefCell::new(Size::ZERO)),
 
-            active_item_index: 0,
             items: Arc::new(Vec::new()),
             diagnostic_content: None,
         }
     }
 
-    pub fn get_current_item(&self) -> Option<&RichText> {
-        self.items.get(self.active_item_index)
+    pub fn get_current_items(&self) -> Option<Vec<&RichText>> {
```

Comment:
If the elements can be references, can we return `Option<&[RichText]>` here, instead? I'm not fond of useless memory allocations. Even better, since you only call `.is_some()` on the result, why not produce the information you actually need?

Reply:
Makes sense! Thanks bugadani, this is my first time writing rust so any feedback is appreciated 🙏